lib/cocoapods/validator.rb

```diff
@@ -28,8 +28,12 @@ class Validator
     #         the Source URLs to use in creating a {Podfile}.
     #
     def initialize(spec_or_path, source_urls)
-      @source_urls = source_urls.map { |url| config.sources_manager.source_with_name_or_url(url) }.map(&:url)
       @linter = Specification::Linter.new(spec_or_path)
+      @source_urls = if !@linter.spec || @linter.spec.dependencies.empty?
```

Comment:
Thinking a bit more about the issue document in #6154, is that even setting an empty --source was going to clone the master repo and that because, like you were saying, the sources methods. So the shortcut is not really needed, a user could set an empty --source and that would work. I'll remove it.